nodestream_plugin_neptune/neptune_connector.py

```diff
@@ -0,0 +1,78 @@
+from nodestream.databases.copy import TypeRetriever
+from nodestream.databases.database_connector import (DatabaseConnector,
+                                                     QueryExecutor)
+
+from .ingest_query_builder import NeptuneIngestQueryBuilder
+from .neptune_connection import NeptuneAnalyticsConnection, NeptuneDBConnection
+from .neptune_query_executor import NeptuneQueryExecutor
+
+
+class NeptuneConnector(DatabaseConnector, alias="neptune"):
+    @classmethod
+    def from_file_data(
+        cls, mode: str, region: str, host: str = None, graph_id: str = None, **kwargs
+    ):
+        return cls(
+            mode=mode,
+            host=host,
+            graph_id=graph_id,
+            region=region,
+            async_partitions=kwargs.get("async_partitions"),
+            ingest_query_builder=NeptuneIngestQueryBuilder(),
+        )
+
+    def __init__(
+        self,
+        mode: str,
+        region: str,
+        async_partitions: int,
+        ingest_query_builder: NeptuneIngestQueryBuilder,
+        host: str = None,
+        graph_id: str = None,
+    ) -> None:
+        if mode != "database" and mode != "analytics":
+            raise ValueError("`mode` must be either 'database' or 'analytics'")
+        if mode == "database":
+            if host is None:
+                raise ValueError(
+                    "A `host` must be specified when `mode` is 'database'."
+                )
+            if graph_id is not None:
+                raise ValueError(
+                    "A `graph_id` should not be used with Neptune Database, `host=<Neptune Endpoint>` should be used instead. If using Neptune Analytics, set `mode='analytics'."
+                )
```

Comment:
If we were to use `NeptuneDBConnection#from_configuration`, we could move the validation there so this method isn't so monolithic. Otherwise we are going to have some [shotgun surgery](https://refactoring.guru/smells/shotgun-surgery) if/when we change the configuration settings. What do you think?

Reply:
Agreed, I've moved the logic into `NeptuneDBConnection#from_configuration`.